packages/util-waiter/src/poller.ts

```diff
@@ -0,0 +1,36 @@
+import { sleep } from "./utils/sleep";
+import { WaiterOptions, WaiterResult, WaiterState } from "./waiter";
+
+/**
+ * Reference: https://github.com/awslabs/smithy/pull/656
+ * The theoretical limit to the attempt is max delay cannot be > Number.MAX_VALUE, but it's unlikely because of
+ * `maxWaitTime`
+ */
+const exponentialBackoffWithJitter = (floor: number, ciel: number, attempt: number) =>
+  Math.floor(Math.min(ciel, randomInRange(floor, floor * 2 ** (attempt - 1))));
```

Comment:
> floor * 2 ** (attempt - 1)

I think this will overflow your number type for large values of attempt. To address this, you can compute the maximum attempt ceiling that doesn't exceed your ceiling.

Pseudo code:

```
attemptCeiling = (log(maxDelay / minDelay) / log(2)) + 1

if attempt > attemptCeiling:
    delay = maxDelay
else:
    delay = minDelay * 2 ** (attempt - 1)

delay = random(minDelay, delay)

if remainingTime - delay <= minDelay:
    delay = remainingTime - minDelay
```

Reply:
This

Reply:
As I mentioned in the comment, this is not possible because it is in fact limited by `maxWaitTime` in the promise race in other abstraction.

Reply:
I dont think Javascript has number overflow: https://www.algotech.solutions/blog/javascript/handle-number-overflow-javascript/

Reply:
@alexforsyth It's possible to overflow the JS Number:
```console
> Number.MAX_VALUE * 2
> Infinity
```